proposals/pytorch-operator-proposal.md

```diff
@@ -0,0 +1,141 @@
+## Motivation
+Pytorch is a popular machine learning framework which currently does not have an operator/controller for Kubernetes. This proposal is aimed at defining what that operator should look like, and adding it to Kubeflow.
```

Comment:
My biggest question about pytorch is how to do distributed communication. Based on my limited research 
kubeflow/kubeflow#179 it looks like there are multiple backends e.g.Gloo and MPI.

Reply:
I've tried it with TCP and Gloo in Kubernetes and it seems to work okay, but I need to build PyTorch with MPI already installed if I want to try it. I also don't have a GPU cluster so I'm probably not seeing the bigger benefits of the different backends. I was thinking just provide the backend as an environment variable and people can choose to use that environment variable in their code if they want or just hard code their backend.

Reply:
I'm gonna have to take back the gloo comment. It's definitely not working right aside from connecting

Reply:
@jose5918 If you need GPUs just let me know you can use dev.kubeflow.org.